lib/pact/consumer/mock_service/app.rb

```diff
@@ -65,14 +72,17 @@ def call env
         response = []
         begin
           relevant_handler = @handlers.detect { |handler| handler.match? env }
-          response = add_cors_header(relevant_handler.respond(env))
+          res= relevant_handler.respond(env)
+          response= relevant_handler.enable_cors? ? add_cors_header(res) : res
         rescue StandardError => e
           @logger.error 'Error ocurred in mock service:'
           @logger.ap e, :error
+          puts e.inspect
```

Comment:
My change is only to fix the typo. Code was pre existing , I don't know if it's needed.